src/http.ts

```diff
@@ -0,0 +1,177 @@
+import * as hrana from "@libsql/hrana-client";
+import { fetch } from "cross-fetch";
+
+import type { Config, Client } from "./api.js";
+import { InStatement, ResultSet, LibsqlError } from "./api.js";
+import { expandConfig } from "./config.js";
+import { stmtToHrana, resultSetFromHrana, mapHranaError } from "./hrana.js";
+
+export * from "./api.js";
+
+export function createClient(config: Config): Client {
+    const expandedConfig = expandConfig(config);
+    const url = expandedConfig.url;
+    if (expandedConfig.transactions) {
+        throw new LibsqlError(
+            "The HTTP client does not support transactions. " +
+            "Please use a libsql:// or libsqls:// URL to allow the client to connect using a WebSocket.",
+            "TRANSACTIONS_NOT_SUPPORTED",
+        );
+    }
+    return new HttpClient(url, expandedConfig.authToken);
+}
+
+export class HttpClient implements Client {
+    #url: URL;
+    #authToken: string | undefined;
+    closed: boolean;
+
+    /** @private */
+    constructor(url: URL, authToken: string | undefined) {
+        this.#url = url;
+        this.#authToken = authToken;
+        this.closed = false;
+    }
+
+    async execute(stmt: InStatement): Promise<ResultSet> {
+        try {
+            const hranaStmt = stmtToHrana(stmt);
+            const protoStmt = hrana.raw.stmtToProto(hranaStmt, true);
+            const request = {"stmt": protoStmt};
+            const response = await this.#send<ExecuteReq, ExecuteResp>("POST", "v1/execute", request);
+            const protoStmtResult = response["result"];
+            const hranaRows = hrana.raw.rowsResultFromProto(protoStmtResult);
+            return resultSetFromHrana(hranaRows);
+        } catch (e) {
+            throw mapHranaError(e);
+        }
+    }
+
+    async batch(stmts: Array<InStatement>): Promise<Array<ResultSet>> {
+        try {
+            const protoSteps: Array<hrana.proto.BatchStep> = [];
+
+            protoSteps.push({
+                "stmt": {"sql": "BEGIN", "want_rows": false},
```

Comment:
Same as with hrana - is sqld managing the transaction details of the batch?

Reply:
Same as with Hrana, because the HTTP API uses the same data structures and semantics as Hrana: sqld provides general batches, the client uses this general mechanism to implement the non-interactive transaction.